llama_index/chat_engine/condense_question.py

```diff
@@ -151,6 +151,14 @@ def chat(
         if self._verbose:
             print(log_str)
 
+        # TODO: right now, query engine uses class attribute to configure streaming,
+        #       we are moving towards separate streaming and non-streaming methods.
+        #       In the meanwhile, use this hack to toggle streaming.
+        from llama_index.query_engine.retriever_query_engine import RetrieverQueryEngine
```

Comment:
Would it be better to check for the base class instead?

Reply:
The `BaseQueryEngine` is an interface that does not enforce having a response synthesizer. So technically, this only works for retriever query engine.

Reply:
aside: this is also why I think having explicit separate non-streaming and streaming endpoints make the behavior more predictable.

Reply:
Ah good point, right